tests/unit/replication_persistance_helper.cpp

```diff
@@ -0,0 +1,199 @@
+                                            "memgraph::storage::replication::ReplicationClientConfig::SSL{{.key_file = "
+                                            "\"{}\", .cert_file = \"{}\"}}",
+                                            status.ssl->key_file, status.ssl->cert_file)
+                                      : "std::nullopt";
+    auto test = fmt::format(
+        "TEST_F(ReplicationPersistanceHelperTest, Test_{}) {{\n"
+        "  auto replicas_status = CreateReplicaStatus(\n"
+        "      \"{}\", \"{}\", {}, memgraph::storage::replication::ReplicationMode::{}, {}, std::chrono::seconds({}),\n"
+        "      {});\n"
+        "\n"
+        "  auto json_status = memgraph::storage::replication::ReplicaStatusToJSON(\n"
+        "      memgraph::storage::replication::ReplicaStatus(replicas_status));\n"
+        "  auto replicas_status_converted = "
+        "memgraph::storage::replication::JSONToReplicaStatus(std::move(json_status));\n"
+        "\n"
+        "  ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));\n"
+        "}}",
+        number_of_tests_failed, name, ip_address, port, sync_mode, timeout, replica_check_frequency, ssl);
+    myfile << test;
+    myfile.close();
+  }
+
+  static_assert(sizeof(memgraph::storage::replication::ReplicaStatus) == 168,
+                "Most likely you modified ReplicaStatus without updating the tests. Please modify CheckEqualityImpl, "
+                "ExportFailingTest and extend the unit tests with relevant tests.");
+};
+
+TEST_F(ReplicationPersistanceHelperTest, BasicTestAllAttributesInitialized) {
+  auto replicas_status = CreateReplicaStatus(
+      "name", "ip_address", 0, memgraph::storage::replication::ReplicationMode::SYNC, 1.0, std::chrono::seconds(1),
+      memgraph::storage::replication::ReplicationClientConfig::SSL{.key_file = "key_file", .cert_file = "cert_file"});
+
+  auto json_status = memgraph::storage::replication::ReplicaStatusToJSON(
+      memgraph::storage::replication::ReplicaStatus(replicas_status));
+  auto replicas_status_converted = memgraph::storage::replication::JSONToReplicaStatus(std::move(json_status));
+
+  ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));
+}
+
+TEST_F(ReplicationPersistanceHelperTest, BasicTestOnlyMandatoryAttributesInitialized) {
+  auto replicas_status =
+      CreateReplicaStatus("name", "ip_address", 0, memgraph::storage::replication::ReplicationMode::SYNC, std::nullopt,
+                          std::chrono::seconds(1), std::nullopt);
+
+  auto json_status = memgraph::storage::replication::ReplicaStatusToJSON(
+      memgraph::storage::replication::ReplicaStatus(replicas_status));
+  auto replicas_status_converted = memgraph::storage::replication::JSONToReplicaStatus(std::move(json_status));
+
+  ASSERT_TRUE(CheckEquality(replicas_status, *replicas_status_converted));
+}
+
+TEST_F(ReplicationPersistanceHelperTest, RandomizedTests) {
+  auto all_tests_succeeded = true;
```

Comment:
```suggestion
```

Reply:
Test is removed, no need for that anymore.